notesapp/api/serializers.py

```diff
@@ -0,0 +1,112 @@
+from rest_framework.serializers import ( 
+	ModelSerializer,
+	HyperlinkedIdentityField,
+	PrimaryKeyRelatedField,
+)
+from notesapp.model import KnowledgeBaseNotes, SoftSkillsDataNotes, RandomDataNotes
+						   
+
+
+knowledgebasenotes_detail_url = HyperlinkedIdentityField(
+    # URL to detail view
+    view_name='knowledgebasenotes-detail',
+    read_only=True
+)
+
+softskillsdatanotes_detail_url = HyperlinkedIdentityField(
+    # URL to detail view
+    view_name='softskillsdatanotes-detail',
+    read_only=True
+)
+
+randomdatanotes_detail_url = HyperlinkedIdentityField(
+    # URL to detail view
+    view_name='randomdatanotes-detail',
+    read_only=True
+)
+
+
+class KnowledgeBaseNotesSerializer(ModelSerializer):
+	"""
+	serializer class for KnowledgeBaseNotes model
+	"""
+	knowledge_base_notes = PrimaryKeyRelatedField(many=True, read_only=True)
```

Comment:
make it PrimaryKeyRelatedField(queryset=KnowledgeBase.objects.all())
because you need to take POST request on this, also many=True not needed here